homeassistant/components/librato.py

```diff
@@ -0,0 +1,115 @@
+        return False
+
+    conf = config[DOMAIN]
+    user = conf[CONF_USER]
+    token = conf[CONF_TOKEN]
+    namespace = conf.get(CONF_NAMESPACE, '')
+    if namespace != '':
+        namespace = namespace + '.'
+
+    metrics = librato.connect(
+        user,
+        token,
+        sanitizer=librato.sanitize_metric_name)
+
+    def metrics_for_states(states):
+        """ Queue metrics for a list of states. """
+        with metrics.new_queue() as metrics_queue:
+            for state in states:
+                # Attempt to parse the state and try to transform it into a
+                # float.
+                _state = None
+                if state.state in (STATE_ON, STATE_LOCKED,
+                                   STATE_ABOVE_HORIZON):
+                    _state = 1
+                elif state.state in (STATE_OFF, STATE_UNLOCKED, STATE_UNKNOWN,
+                                     STATE_BELOW_HORIZON):
+                    _state = 0
+                else:
+                    _state = state.state
+                    try:
+                        _state = float(_state)
+                    except ValueError:
+                        _state = None
+
+                    if _state and isinstance(_state, int):
+                        _state = float(_state)
+
+                if _state is not None:
+                    metrics_queue.add(
+                        '{}states.{}'.format(namespace, state.entity_id),
+                        _state)
+
+                # Now inspect the attributes and submit anything that looks
+                # like a float
+                attributes = dict(state.attributes)
+                ignored_attributes = [
+                    'hidden',
+                    'can_cancel',
+                    'media_title',
+                    'supported_media_commands',
+                    'node_id',
+                    'auto'
+                ]
+                for key in attributes:
+                    if key in ignored_attributes:
+                        continue
+                    try:
+                        value = float(attributes[key])
+                        metrics_queue.add(
+                            '{}states.{}.{}'.format(
+                                namespace,
+                                state.entity_id,
+                                key),
+                            value)
+                    except (TypeError, ValueError):
+                        pass
+
+    def metrics_listener(event):
+        """ Listen for new messages on the bus and send metrics to Librato. """
+        state = event.data.get('new_state')
+        metrics_for_states([state])
+    hass.bus.listen(EVENT_STATE_CHANGED, metrics_listener)
+
+    # pylint: disable=unused-argument
+    def periodic_metrics(now):
+        """ Load all states and submit metrics. """
+        metrics_for_states(hass.states.all())
+    track_utc_time_change(hass, periodic_metrics, second=60)
```

Comment:
Second will never be 60. After 59 comes to 0. If you want to run it every minute, pick any number between 0…59. Whenever the time matches that second, it will fire.

That being said, what is the added value of adding more data points like this ? I would expect this to skew your results.

Imagine at 1:00 you report the temperature as 20 and at 2:00 you report the temperature as 40. If you report the data every minute, it will look like the temperature jumped 20 degrees between reports 1:59 and 2:00 while it probably gradually improved over time (which could be interpolated if only state changes are reported).